### ISSUE TITLE
Electrum Server is reporting "failed to connect daemon at 127.0.0.1:8332: Too many open files (os error 24)"

### ISSUE THREAD

**Issue by HamishMacEwan** [mynodebtc/mynode#439]
**Describe the bug**
electrs is reporting problems connecting to bitcoind:

ELECTRUM SERVER STATUS
 Running
-- Logs begin at Sat 2020-11-21 22:41:03 NZDT, end at Sun 2020-11-22 15:34:20 NZDT. --
Nov 22 15:34:20 myNode electrs[3410]: 2020-11-22T15:34:20.512+13:00 - WARN - failed to connect daemon at 127.0.0.1:8332: Too many open files (os error 24)
Nov 22 15:34:17 myNode electrs[3410]: 2020-11-22T15:34:17.511+13:00 - WARN - failed to connect daemon at 127.0.0.1:8332: Too many open files (os error 24)
Nov 22 15:34:14 myNode electrs[3410]: 2020-11-22T15:34:14.511+13:00 - WARN - failed to connect daemon at 127.0.0.1:8332: Too many open files (os error 24) 

The message repeats every three or so seconds... 

Consequently electrs is continuously "Syncing... 99.96%" and will not accept connections from Electrum Desktop

**Expected behavior**

That electr gets connected to bitcoind on port 8332 and gets the information to complete its synchronisation and accepts connections, as it has before, from Electrum Desktop

**Screenshots**
NA

**Desktop (please complete the following information):**
 - Chrome Version 86.0.4240.193 (Official Build) (64-bit)
 - Microsoft Windows 10 Home Version	10.0.19041 Build 19041

**MyNode hardware (please complete the following information):**
 - Device: Raspi4 4G ram
 - Version 0.2.16

**Additional context**
NA

**Comment by HamishMacEwan** [comment]
Hunting Google I found this:

[https://github.com/romanz/electrs/issues/109](https://github.com/romanz/electrs/issues/109)

And while that is inconclusive I have followed the advice of running `lsof` and have a 110M of results as root, and a smaller file as user bitcoin.

Disabling and re-enabling ELECTRUM SERVER has restored functionality.

Since I'm currently performing an IBD using the mynode.local device as an `addnode` peer, this may have caused the problem.

**Comment by HamishMacEwan** [comment]
The issue has returned, and after disable/enable electrs the log is full of this section repeating:

```
Nov 23 12:38:20 myNode electrs[20761]: Caused by: Broken pipe (os error 32)
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.812+13:00 - ERROR - [127.0.0.1:38940] connection handling failed: Error: failed to send {"id":32,"jsonrpc":"2.0","result":null}
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.769+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 32, \"params\": [\"33d833fd6dcbef6f5d7a5fc49eef7e9b54939260537bd5ca796dedbce514b1ce\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.665+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 31, \"params\": [\"d9f4dc707b19d8064106b34ca4f9aefd3682f116e2d1fdcfa7f29bdbc2cc8c70\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.630+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 30, \"params\": [\"0506658d83d56c62db99e8365fc5575eb79ba0ec64512538fec77fde575c7e81\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.523+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 29, \"params\": [\"64b8cc58751882ed827c0a268c44307e48d003aa8651c7f30ce75ca4ffa1b968\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.485+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 28, \"params\": [\"edbcf0f54111e72c53fc5ac2d56d8cc18ae00138b65b5070fd739a8cdd45682a\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.473+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 27, \"params\": [\"650f95fb7b5868085b4975b88eef5bbf00ddfac9a38d90aab3a9de9292140ae9\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.452+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 26, \"params\": [\"a7e0f803c53a28f96020f5e90a39f20c4184f8a3a89dc8bdf862c075c8720c78\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.426+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 25, \"params\": [\"a71768490f397afffd322b5a2da7c15c598990f35137c9c9de0aa721d4cd5af8\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.389+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 24, \"params\": [\"83b7d2d219005f6cfc0eee0f369d8a78c752c449e73c295e3d097204ad127f9a\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.389+13:00 - TRACE - RPC PeriodicUpdate
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.220+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 23, \"params\": [\"bd480b9bf76e9155655dd82d857f56c361070c28390b177a6cecf725b790628f\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.111+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 22, \"params\": [\"7d6a883b3e49b532007035cae1c3025462c5712cfe0d1623395c73c49fed6acb\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.086+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 21, \"params\": [\"73396b55bd7d01839c788f096bb9e2b282eefcd3b86cc03f315eca13d3fa7522\"]}\n")
Nov 23 12:38:20 myNode electrs[20761]: 2020-11-23T12:38:20.086+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 20, \"params\": [\"5a0c5a6641e3295ab1a597f531a808e761e30bd45591b58d54c4fe3e39bad24f\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.918+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 19, \"params\": [\"72ce0f5be0f149fc1f8dcdb00e1bf5effd5706497918529df88d69e25d102fa4\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.897+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 18, \"params\": [\"c61d0cb22a295bc313cff4dd5dc45e35867c1329285eb4270cba32f0a4d3fed7\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.870+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 17, \"params\": [\"9147ab776e018acf2014cdf56ddef8555940f4020f5b7382577c4a25b3b46bb5\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.834+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 16, \"params\": [\"b40dd04c64afe7c6536392f27a61918d6ffd93577e336f8409ab054c77583460\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.754+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 15, \"params\": [\"e0f9060a21f04229eb5cdfbf28cef61cb148d7800ebcac1112354698cf7759d3\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.721+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 14, \"params\": [\"19cff2d8070b4a00ce2c498dd383118bab7b2c83bdf96491baf262f7a35cc255\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.628+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 13, \"params\": [\"c99e1fa0144a78623a8b234ecd928d32400a7c8fad9e2e8de795916c4b2f4408\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.585+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.scripthash.subscribe\", \"id\": 12, \"params\": [\"238a42ebd843e987e82ebfa3bf46d1e49d20748b757d6cabc1f5c8bccaab314c\"]}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.585+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"mempool.get_fee_histogram\", \"id\": 11}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.585+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.relayfee\", \"id\": 10}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.584+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"server.peers.subscribe\", \"id\": 9}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.584+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"server.donation_address\", \"id\": 8}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.582+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"server.banner\", \"id\": 7}\n")
Nov 23 12:38:19 myNode electrs[20761]: 2020-11-23T12:38:19.577+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.block.header\", \"id\": 6, \"params\": [658202]}\n")
Nov 23 12:38:18 myNode electrs[20761]: 2020-11-23T12:38:18.832+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.estimatefee\", \"id\": 5, \"params\": [2]}\n")
Nov 23 12:38:18 myNode electrs[20761]: 2020-11-23T12:38:18.832+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.estimatefee\", \"id\": 4, \"params\": [5]}\n")
Nov 23 12:38:18 myNode electrs[20761]: 2020-11-23T12:38:18.831+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.estimatefee\", \"id\": 3, \"params\": [10]}\n")
Nov 23 12:38:18 myNode electrs[20761]: 2020-11-23T12:38:18.831+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.estimatefee\", \"id\": 2, \"params\": [25]}\n")
Nov 23 12:38:18 myNode electrs[20761]: 2020-11-23T12:38:18.828+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"blockchain.headers.subscribe\", \"id\": 1}\n")
Nov 23 12:38:17 myNode electrs[20761]: 2020-11-23T12:38:17.766+13:00 - TRACE - RPC Request("{\"jsonrpc\": \"2.0\", \"method\": \"server.version\", \"id\": 0, \"params\": [\"electrum/4.0.5\", \"1.4\"]}\n")
Nov 23 12:38:17 myNode electrs[20761]: 2020-11-23T12:38:17.436+13:00 - INFO - [127.0.0.1:38940] connected peer
Nov 23 12:38:17 myNode electrs[20761]: 2020-11-23T12:38:17.436+13:00 - TRACE - [127.0.0.1:38940] spawned ThreadId(46)
Nov 23 12:38:14 myNode electrs[20761]: 2020-11-23T12:38:14.849+13:00 - DEBUG - [127.0.0.1:38936] shutting down connection
Nov 23 12:38:14 myNode electrs[20761]: Caused by: Broken pipe (os error 32)
```
These appear to be requests from Electrum Desktop as the status changes from red dot, to the whirl (two arrows) icon, and back to red in synch with these errors.

**Comment by Jacksper13** [comment]
@tehelsper new image available for download, this issue was on v0.2.16 - I'd close issue
